Source/MQTTnet.AspnetCore/MqttConnectionContext.cs

```diff
@@ -211,8 +211,17 @@ public async Task SendPacketAsync(MqttPacket packet, CancellationToken cancellat
                 {
                     var buffer = PacketFormatterAdapter.Encode(packet);
 
-                    WritePacketBuffer(_output, buffer);
-                    await _output.FlushAsync(cancellationToken).ConfigureAwait(false);
+                    if (buffer.Payload.Count == 0)
+                    {
+                        // zero copy
+                        // https://github.com/dotnet/runtime/blob/main/src/libraries/System.IO.Pipelines/src/System/IO/Pipelines/StreamPipeWriter.cs#L279
```

Comment:
Should the commit be given instead the branch _main_?
I mean the code can change, so this link can become a phantom.
```suggestion
                        // https://github.com/dotnet/runtime/blob/e31ddfdc4f574b26231233dc10c9a9c402f40590/src/libraries/System.IO.Pipelines/src/System/IO/Pipelines/StreamPipeWriter.cs#L279
```